pkg/auth/service.go

```diff
@@ -959,3 +964,632 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
 	// we're allowed!
 	return &AuthorizationResponse{Allowed: true}, nil
 }
+
+type APIAuthService struct {
+	apiClient   *ClientWithResponses
+	secretStore crypt.SecretStore
+	cache       Cache
+}
+
+func (a *APIAuthService) SecretStore() crypt.SecretStore {
+	return a.secretStore
+}
+
+func (a *APIAuthService) CreateUser(ctx context.Context, user *model.User) (int64, error) {
+	resp, err := a.apiClient.CreateUserWithResponse(ctx, CreateUserJSONRequestBody{
+		Username: user.Username,
+	})
+	if err != nil {
+		return InvalidUserID, err
+	}
+	if err := validateResponse(resp, 201); err != nil {
+		return InvalidUserID, err
+	}
+
+	return resp.JSON201.Id, nil
```

Comment:
`resp.json` could not be nil In case the return status code is 201 which is validated in `validateResponse`
@nopcoder Do you think I should add this anyway?

Reply:
not for this pr 👍